src/operator/keycloak.ts

```diff
@@ -451,67 +415,67 @@ async function keycloakRealmProviderConfigurer(api: KeycloakApi) {
     env.IDP_GROUP_PLATFORM_ADMIN,
     env.KEYCLOAK_REALM,
   )
-  const existingRealmRoles = ((await doApiCall(errors, `Getting all roles from realm ${keycloakRealm}`, async () =>
-    api.roles.realmRolesGet(keycloakRealm),
-  )) || []) as Array<RealmRole>
+  console.info(`Getting all roles from realm ${keycloakRealm}`)
+  const existingRealmRoles = ((await api.roles.realmRolesGet(keycloakRealm)).body || []) as RealmRole[]
```

Comment:
In fact I don't know if the empty-default is even necessary. I do not see a good reason why it would return an empty body – it should either be an empty list or maybe a 404 error if the realm does not exist. But again, this part I only migrated syntactically, so I did not check all possible scenarios of how the Keycloak API works.

Reply:
Here we do a fallback to an empty array. So if the array is empty we can immediately create the role right?
But as you said below I think this is an old fallback and now the api call will always return something that is correct or an error. So we could also remove this fallback.

Reply:
Yes, I now tested it. I dropped all the fallbacks in the code, deleted the Keycloak DB in a running test cluster, and had it rebuilt from scratch. That way all instances should have encountered an empty list at least once; there were no errors.